winner-tool/src/main.rs

```diff
@@ -0,0 +1,179 @@
+//! This tool calculates the quantitative category winners for Tour de SOL.
+//!
+//! NOTE: Ledger processing uses native programs, so this tool must be invoked with `cargo run`.
+//! If installed with `cargo install` the native programs may not be linked properly.
+
+mod availability;
+mod confirmation_latency;
+mod rewards_earned;
+mod utils;
+mod winner;
+
+use clap::{crate_description, crate_name, crate_version, value_t, value_t_or_exit, App, Arg};
+use confirmation_latency::{SlotVoterSegments, VoterRecord};
+use solana_cli::input_validators::is_pubkey;
+use solana_core::blocktree::Blocktree;
+use solana_core::blocktree_processor::{process_blocktree, ProcessOptions};
+use solana_runtime::bank::Bank;
+use solana_sdk::genesis_block::GenesisBlock;
+use solana_sdk::native_token::sol_to_lamports;
+use solana_sdk::pubkey::Pubkey;
+use std::collections::HashSet;
+use std::path::PathBuf;
+use std::process::exit;
+use std::str::FromStr;
+use std::sync::{Arc, RwLock};
+
+fn main() {
+    solana_logger::setup();
+
+    let matches = App::new(crate_name!())
+        .about(crate_description!())
+        .version(crate_version!())
+        .arg(
+            Arg::with_name("ledger")
+                .short("l")
+                .long("ledger")
+                .value_name("DIR")
+                .takes_value(true)
+                .required(true)
+                .help("Use directory for ledger location"),
+        )
+        .arg(
+            Arg::with_name("starting_balance")
+                .long("starting-balance")
+                .value_name("SOL")
+                .takes_value(true)
+                .default_value("1000")
+                .help("Starting balance of validators at the beginning of TdS"),
+        )
+        .arg(
+            Arg::with_name("baseline_validator")
+                .long("baseline-validator")
+                .value_name("PUBKEY")
+                .takes_value(true)
+                .required(true)
+                .validator(is_pubkey)
+                .help("Public key of the baseline validator"),
+        )
+        .arg(
+            Arg::with_name("bootstrap_leader")
+                .long("bootstrap-leader")
+                .value_name("PUBKEY")
+                .takes_value(true)
+                .required(true)
+                .validator(is_pubkey)
+                .help("Public key of the bootstrap leader"),
+        )
+        .arg(
+            Arg::with_name("final_slot")
+                .long("final-slot")
+                .value_name("SLOT")
+                .takes_value(true)
+                .help("Final slot of TdS ledger"),
+        )
+        .get_matches();
+
+    let ledger_path = PathBuf::from(value_t_or_exit!(matches, "ledger", String));
+    let starting_balance_sol = value_t_or_exit!(matches, "starting_balance", f64);
+    let baseline_id_string = value_t_or_exit!(matches, "baseline_validator", String);
```

Comment:
Does this work?  `let baseline_id_string = value_t_or_exit!(matches, "baseline_validator", Pubkey);`

Reply:
Yes, fixed here: https://github.com/solana-labs/tour-de-sol/pull/15/files#diff-69c184a9c7bce2c16d29851280d961faR82